### PR TITLE
Replace react-popper usage in terra-date-picker with terra-hookshot

### LINKED ISSUE #597: Update date picker to use hookshot for positioning
# Feature Request

## Description
Now that we've forked the date picker we were using, we should look into updating the positioning logic to use terra-hookshot instead of react-popper.

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Resolves #597

### CODE REVIEW

File: packages/terra-date-picker/src/DatePositionUtil.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
+ * Calculates the bottom position of the element.
```
Comment (bjankord) [inline packages/terra-date-picker/src/DatePositionUtil.js]: This is based on [logic we use for positioning arrows in terra-popup](https://github.com/cerner/terra-framework/blob/master/packages/terra-popup/src/_PopupUtils.js#L187).

File: packages/terra-date-picker/src/react-datepicker/index.jsx [hunk 2]
```diff
@@ -542,19 +550,31 @@ export default class DatePicker extends React.Component {
...
-          <div className="react-datepicker__input-container">
-            {this.renderDateInput()}
-            {this.renderClearButton()}
-          </div>
-        }
-        popperContainer={this.props.popperContainer}
-        popperComponent={calendar}
-        popperPlacement={this.props.popperPlacement}/>
+      <React.Fragment>
+        <div ref={(container) => { this.datePickerContainer = container; }} className="react-datepicker__input-container">
```
Comment (tbiethman) [inline packages/terra-date-picker/src/react-datepicker/index.jsx]: Not a huge deal, but it'd be nice to start using `createRef` for our ref needs: https://reactjs.org/docs/refs-and-the-dom.html#creating-refs

**Comment by mmalaker** [comment]
Functional verification completed. The date picker examples display and function as expected using Hookshot for positioning.

File: packages/terra-date-picker/src/react-datepicker/index.jsx (hunk 2 above)
Comment (bjankord) [reply packages/terra-date-picker/src/react-datepicker/index.jsx]: Good call, updated in this commit: https://github.com/cerner/terra-framework/pull/601/commits/dea77f089b4c8d6dd9e45945272ad2e9d3101423

**Pull request merged**